setup.py

```diff
@@ -19,23 +42,64 @@
         "tests/data/*.txt",
         "*.yaml",
     ],
+
     "jwst.fits_generator": [
         "templates/*.inc",
         "templates/*.txt",
         "tests/okfile/*.prop",
     ],
+
     "jwst.lib": ["tests/data/siaf.db"],
+
     # Include the rules .py files in associations test data
     "jwst.associations": ["tests/data/*.py"],
+
+    # Include C extensions
+    "jwst.lib.src": ["*.c"],
+
     # Include the transforms schemas
     "jwst.transforms": ["schemas/stsci.edu/jwst_pipeline/*.yaml"],
     "jwst.stpipe.resources": ["schemas/*.yaml"],
 }
 
+# Install packages required for this setup to proceed:
+_install_setup_requires(dict(setup_requires=SETUP_REQUIRES))
+for dep_pkg in SETUP_REQUIRES:
+    try:
+        importlib.import_module(dep_pkg)
+    except ImportError:
+        print(f"{dep_pkg} is required in order to install '{PACKAGENAME}'.\n"
+              f"Please install {dep_pkg} first.",
+              file=sys.stderr)
+        exit(1)
+
+# Setup C module include directories
+import numpy
+include_dirs = [numpy.get_include()]
+
+# Setup C module macros
+define_macros = [('NUMPY', '1')]
+
+# Handle MSVC `wcsset` redefinition
+if sys.platform == 'win32':
+    define_macros += [
+        ('_CRT_SECURE_NO_WARNING', None),
+        ('__STDC__', 1)
+    ]
+
```

Comment:
I believe none of this except `include_dirs` is needed, as again, pyproject.toml assures that `numpy` is installed before `setup()` is run.  Also we don't support Windows (currently).

Reply:
@jdavies-st OK. Removed support for installing/compiling on Windows and support for installing using `python setup.py install` in https://github.com/spacetelescope/jwst/pull/5978/commits/a9de43db2bba745cd793f57d20023c7d1b58477a